src/OSPSuite.Core/Domain/Services/ParameterIdentifications/IParameterIdentificationTask.cs

```diff
@@ -80,5 +80,11 @@ public interface IParameterIdentificationTask
       /// Returns a clone of the <paramref name="parameterIdentification"/> or null if the user cancels the clone action
       /// </summary>
       ParameterIdentification Clone(ParameterIdentification parameterIdentification);
+
+      /// <summary>
+      /// Updates the <see cref="ParameterIdentification"/>s using the <paramref name="observedData"/>
+      /// Returns an Ienumerable of <see cref="OutputMapping"/> that have been updated
+      /// </summary>
```

Comment:
This doesn't return an Ienumerable, and our convention is to not use IEnumerable as argument, but IReadOnlyList instead. That ensures the enumeration is only evaluated only once.

You could maybe change the name of this method to `ResetOutputMappingWeightFor` or something like that. IMO it needs to be explained in the summary that this is going to reset the weights if the number of rows has changed, otherwise weights are not affected